### PR TITLE
feat: Enable GitHub actions.

### PR DESCRIPTION
**Pull request by ravali-rimmalapudi**
# Fixes #
Closes: [[DII-106](https://issues.corp.twilio.com/browse/DII-106), [DII-107](https://issues.corp.twilio.com/browse/DII-107)]
Dependent PR: [CLI](https://github.com/twilio/twilio-cli/pull/284)
Enable the GitHub actions.
Changes:
1. Added workflow file
2. Added scripts to update the changelog in cli-core
3. Added the actions to bump the version and release to GitHub

Testing Gist: [Link](https://gist.github.com/ravali-rimmalapudi/09ec56097282c88ab8087b61640edbf0)

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [ ] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-cli-core/blob/main/CONTRIBUTING.md) and my PR follows them
- [ ] I have titled the PR appropriately
- [ ] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://twilio.com/help/contact), or create a GitHub Issue in this repository.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+          npm ci
+          npm test
+  update-api-specs:
+    runs-on: ubuntu-latest
+    needs: [test]
+    outputs:
+      change-log: ${{ steps.update-specs.outputs.change-log }}
+      version-type: ${{ steps.update-specs.outputs.version-type }}
+    steps:
+      - name: Create temporary folder for copying json files fom OAI repo
```
Comment (ravali-rimmalapudi) [inline .github/workflows/release.yml:34]: Typo: `fom`

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+          git pull
+          npm install
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: 'npm'
+      - name: semanticRelease
+        run: |
+          npm install --save-dev @semantic-release/changelog @semantic-release/git
```
Comment (ravali-rimmalapudi) [inline .github/workflows/release.yml:70]: Will remove this.

File: package.json [hunk 3]
```diff
@@ -59,5 +59,41 @@
   },
   "engines": {
     "node": ">=10.12.0"
+  },
```
Comment (ravali-rimmalapudi) [inline package.json:62]: Add `@semantic-release/changelog @semantic-release/git` dev dependency.

File: .github/scripts/change-log-helper.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    oaiChangelogFilename = oaiChangelogFile,
+    versionRegex = defaultVersionRegex,
+    dateRegex = defaultDateRegex,
+  ) {
+    this.versionRegex = versionRegex;
+    this.dateRegex = dateRegex;
+    this.cliCoreChangelogFilename = cliCoreChangelogFilename;
+    this.oaiChangelogFilename = oaiChangelogFilename;
+    this.logger = logger;
+  }
```
Comment (onuzbee) [inline .github/scripts/change-log-helper.js:23]: Plans for UTs for these code snippets?

Comment (ravali-rimmalapudi) [reply .github/scripts/change-log-helper.js:23]: Thanks. Added the test cases.

**Review by onuzbee — APPROVED** [review]
Good work! 🚀

**Pull request merged**